sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -363,15 +363,19 @@ public Decimal getDecimal(int ordinal, int precision, int scale) {
   @Override
   public UTF8String getUTF8String(int ordinal) {
     assertIndexIsValid(ordinal);
-    return isNullAt(ordinal) ? null : UTF8String.fromBytes(getBinary(ordinal));
```

Comment:
Yes, It's more risky now, but we already do this for StructType, may do this for ArrayType and MapType.

We need to pay more attentions to it, and fix them case by case.